pallets/pallet-validator/src/lib.rs

```diff
@@ -0,0 +1,334 @@
+		ValidatorActivated(types::Validator<T::AccountId>),
+		ValidatorRemoved(types::Validator<T::AccountId>),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		AlreadyBonded,
+		StashNotBonded,
+		StashBondedWithWrongValidator,
+		CannotBondWithSameAccount,
+		DuplicateValidator,
+		ValidatorNotFound,
+		ValidatorNotApproved,
+		UnauthorizedToActivateValidator,
+		ValidatorValidatingAlready,
+		ValidatorNotValidating,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub validators: Vec<T::AccountId>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				validators: Vec::new(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a request to become a validator
+		/// Validator account (signer): the account of the validator (this account will be added to the council)
+		/// Validator node account: the account that will validate on consensus layer
+		/// Stash account: the "bank" account of the validator (where rewards should be sent to) the stash should be bonded to a validator
+		/// Description: why someone wants to become a validator
+		/// Tf Connect ID: the threefold connect ID of the persion who wants to become a validator
+		/// Info: some public info about the validator (website link, blog link, ..)
+		/// A user can only have 1 validator request at a time
+		#[pallet::weight(100_000_000)]
+		pub fn create_validator(
+			origin: OriginFor<T>,
+			validator_node_account: T::AccountId,
+			stash_account: T::AccountId,
+			description: Vec<u8>,
+			tf_connect_id: u64,
+			info: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin.clone())?;
+
+			// Request should not be a duplicate
+			ensure!(
+				!<Validator<T>>::contains_key(&address),
+				Error::<T>::DuplicateValidator
+			);
+
+			// Request stash account should be bonded
+			Self::check_bond(&stash_account, &address)?;
+
+			let request = types::Validator {
+				validator_node_account: validator_node_account.clone(),
+				stash_account,
+				description,
+				tf_connect_id,
+				info,
+				state: types::ValidatorRequestState::Created,
+			};
+
+			// Create a validator request object
+			<Validator<T>>::insert(&address, &request);
+
+			Self::deposit_event(Event::ValidatorCreated(address, request.clone()));
+
+			Ok(().into())
+		}
+
+		/// Start participating in consensus
+		/// Will activate the Validator node account on consensus level
+		/// A user can only call this if his request to be a validator is approved by the council
+		/// Should be called when his node is synced and ready to start validating
+		#[pallet::weight(100_000_000)]
+		pub fn activate_validator(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
```

Comment:
The validator is already active as it can already vote in the DAO council, it should be `activate_validator_node`.